web/src/utils/helpers.ts

```diff
@@ -81,35 +82,47 @@ export function useNavigateWithParameters() {
 
   const basePath = isZoneRoute ? '/zone' : '/map';
 
-  return ({
-    to = basePath,
-    zoneId = isZoneRoute ? previousZoneId : undefined,
-    timeRange = previousTimeRange,
-    datetime = previousDatetime,
-    keepHashParameters = true,
-  }: {
-    to?: string;
-    zoneId?: string;
-    timeRange?: string;
-    datetime?: string;
-    keepHashParameters?: boolean;
-  }) => {
-    // Always preserve existing search params
-    const isDestinationZoneRoute = to.startsWith('/zone');
-    const currentSearch = new URLSearchParams(location.search);
-    const path = getDestinationPath({
-      to,
-      zoneId: isDestinationZoneRoute ? zoneId : undefined,
-      timeRange,
-      datetime,
-    });
-    const fullPath = {
-      pathname: path,
-      search: currentSearch.toString() ? `?${currentSearch.toString()}` : '',
-      hash: keepHashParameters ? location.hash : undefined,
-    };
-    navigator(fullPath);
-  };
+  return useCallback(
+    ({
+      to = basePath,
+      zoneId = isZoneRoute ? previousZoneId : undefined,
+      timeRange = previousTimeRange,
+      datetime = previousDatetime,
+      keepHashParameters = true,
+    }: {
+      to?: string;
+      zoneId?: string;
+      timeRange?: string;
+      datetime?: string;
+      keepHashParameters?: boolean;
+    }) => {
+      // Always preserve existing search params
+      const isDestinationZoneRoute = to.startsWith('/zone');
+      const currentSearch = new URLSearchParams(location.search);
+      const path = getDestinationPath({
+        to,
+        zoneId: isDestinationZoneRoute ? zoneId : undefined,
+        timeRange,
+        datetime,
+      });
+      const fullPath = {
+        pathname: path,
+        search: currentSearch.toString() ? `?${currentSearch.toString()}` : '',
+        hash: keepHashParameters ? location.hash : undefined,
+      };
+      navigator(fullPath);
+    },
+    [
+      basePath,
+      isZoneRoute,
+      location.hash,
+      location.search,
+      navigator,
+      previousDatetime,
+      previousTimeRange,
+      previousZoneId,
+    ]
+  );
```

Comment:
Wrapping this into a `useCallback` is a best practice:
According to [this](https://react.dev/reference/react/useCallback#optimizing-a-custom-hook):
> If you’re writing a [custom Hook,](https://react.dev/learn/reusing-logic-with-custom-hooks) it’s recommended to wrap any functions that it returns into useCallback